### PR TITLE
RE-1306 Detect mainline=rc and fail out

### PR DESCRIPTION
**Pull request by hughsaunders**
This patch also makes the default command more explicit so that
its easier to predict the effect changing each variable will have.

Issue: [RE-1306](https://rpc-openstack.atlassian.net/browse/RE-1306)

### CODE REVIEW

File: lint.sh [hunk 1]
```diff
@@ -17,7 +17,7 @@ install(){
     virtualenv $venv
   fi
   . $venv/bin/activate
-  pip install -c constraints.txt -r test-requirements.txt >/dev/null
+  pip install -c constraints.txt -r test-requirements.txt -r requirements.txt >/dev/null
```
Comment (mattt416) [inline lint.sh]: @hughsaunders I think we determined yesterday that this isn't needed by the gates.  Can you please remove this change?

Comment (hughsaunders) [reply lint.sh]: Done

File: scripts/ghutils.py [hunk 2]
```diff
@@ -170,6 +170,13 @@ def update_rc_branch(ctx, mainline, rc):
...
     """
+
+    if mainline == rc:
+        raise ValueError("Specifying the same branch for mainline and rc"
+                         " will result in dataloss. The mainline branch"
+                         " will be deleted, then the rc branch will be"
+                         " created from the now non-existent mainline branch")
+
     repo = ctx.obj
     rc = try_context(repo, rc, "rc", "rc_ref")
```
[the remark this answers is not in the document]
Comment (hughsaunders) [reply scripts/ghutils.py]: Done

**Comment by hughsaunders** [comment]
Changes since previous reviews:
```diff
10:24 $ git diff origin/RE-1306_1 origin/RE-1306-3
diff --git a/lint.sh b/lint.sh
index 2787e2e..4f84dd4 100755
--- a/lint.sh
+++ b/lint.sh
@@ -17,7 +17,7 @@ install(){
     virtualenv $venv
   fi
   . $venv/bin/activate
-  pip install -c constraints.txt -r test-requirements.txt -r requirements.txt >/dev/null
+  pip install -c constraints.txt -r test-requirements.txt >/dev/null
 }

 create_jjb_ini(){
diff --git a/scripts/ghutils.py b/scripts/ghutils.py
index ece471d..eecd97f 100644
--- a/scripts/ghutils.py
+++ b/scripts/ghutils.py
@@ -3,13 +3,14 @@
 # This script contains github related utilties for Jenkins.


-import click
-import git
-import github3
 import json
 import logging
 import re

+import click
+import git
+import github3
+
 from notifications import try_context

 logger = logging.getLogger("ghutils")
@@ -170,6 +171,8 @@ def update_rc_branch(ctx, mainline, rc):
     3. Create rc branch from head of mainline
     4. Enable branch protection with skeleton or previously stored settings.
     """
+    repo = ctx.obj
+    rc = try_context(repo, rc, "rc", "rc_ref")

     if mainline == rc:
         raise ValueError("Specifying the same branch for mainline and rc"
@@ -177,8 +180,6 @@ def update_rc_branch(ctx, mainline, rc):
                          " will be deleted, then the rc branch will be"
                          " created from the now non-existent mainline branch")

-    repo = ctx.obj
-    rc = try_context(repo, rc, "rc", "rc_ref")
     branch_protection_enabled = False

     # check if branch exists
```

**Comment by git-harry** [comment]
recheck_xenial_functional_test
recheck_unit_tests

**Pull request merged**